### PR TITLE
Close database and assert afterwards on the result.

### PR DESCRIPTION
**Pull request by BenchR267**
rdar://55274300

### CODE REVIEW

**Comment by BenchR267** [comment]
@swift-ci please smoke test

File: lib/Core/SQLiteBuildDB.cpp [hunk 1]
```diff
@@ -327,7 +327,8 @@ class SQLiteBuildDB : public BuildDB {
     sqlite3_finalize(getKeysWithResultStmt);
     getKeysWithResultStmt = nullptr;
 
-    assert(sqlite3_close(db) == SQLITE_OK && "The database connection could not be closed. That means there are prepared statements that are not finalized, data blobs that are not closed or backups not finished.");
+    int result = sqlite3_close(db);
```
Comment (ddunbar) [inline lib/Core/SQLiteBuildDB.cpp]: You might need a ‘(void)result’ line after this to prevent warnings in no assert builds.

Comment (BenchR267) [reply lib/Core/SQLiteBuildDB.cpp]: Yep, you're right. Thanks

**Comment by BenchR267** [comment]
@swift-ci please smoke test

**Comment by BenchR267** [comment]
@swift-ci please smoke test

**Comment by BenchR267** [comment]
@swift-ci please smoke test

**Comment by BenchR267** [comment]
@swift-ci please smoke test

**Pull request merged**